src/lib/comms/include/sol-http-server.h

```diff
@@ -56,25 +56,126 @@ extern "C" {
 struct sol_http_server;
 struct sol_http_request;
 
+/**
+ * It creates a http's server on that will bind on all interfaces in the given port.
+ * With the returned handle it's possible to register paths using @c sol_http_server_register_handler
+ * and dirs with @c sol_http_server_add_dir.
+ *
+ * @note Only one instance of sol_http_server is possible per port. Try to run a second instance in the
+ * same port will result in fail.
+ *
+ * @param port The port where the server will bind.
+ *
+ * @return a handle to the server on success, otherwise @c NULL is returned.
+ */
 struct sol_http_server *sol_http_server_new(uint16_t port);
+
+/**
+ * Deletes the server and consequently all handlers and dir added.
```

Comment:
I know, but the documentation is dubious and scary-sounding. I'd just say that this destroys the server instance, leaving the removal of handlers and directory references implicit.